refinery/ui/source/js/data-set-import/controllers/file-upload-s3.js

```diff
@@ -5,25 +5,134 @@
     .module('refineryDataSetImport')
     .controller('RefineryFileUploadS3Ctrl', RefineryFileUploadS3Ctrl);
 
-  RefineryFileUploadS3Ctrl.$inject = ['s3UploadService'];
+  RefineryFileUploadS3Ctrl.$inject = ['$log', '$scope', 's3UploadService'];
 
-  function RefineryFileUploadS3Ctrl (s3UploadService) {
+  function RefineryFileUploadS3Ctrl ($log, $scope, s3UploadService) {
     var vm = this;
-    vm.uploadFiles = function (files) {
-      vm.files = files;
-      if (files && files.length > 0) {
-        angular.forEach(vm.files, function (file) {
-          s3UploadService.upload(file).then(function () {
-            // Mark as success
-            file.success = true;
-          }, function (error) {
-            // Mark the error
-            vm.error = error;
-          }, function (progress) {
-            // Write the progress as a percentage
+    vm.multifileUploadInProgress = false;
+
+    vm.isUploadConfigured = function () {
+      if (s3UploadService.isConfigReady) {
+        return s3UploadService.isConfigValid();
+      }
+      return false;
+    };
+
+    vm.addFiles = function (files) {
+      if (typeof vm.files === 'undefined') {
+        vm.files = files;
+      } else {
+        vm.files = vm.files.concat(files);
+      }
+    };
+
+    vm.isFileNew = function (file) {
+      // check if file upload has been attempted
+      return typeof file.progress === 'undefined';
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined

"However, this kind of technique should be avoided. JavaScript is a statically scoped language, so knowing if a variable is declared can be read by seeing whether it is declared in an enclosing context. The only exception is the global scope, but the global scope is bound to the global object, so checking the existence of a variable in the global context can be done by checking the existence of a property on the global object (using the in operator, for instance)."